### PR TITLE
website: remove legacy provider docs index

### PR DESCRIPTION
**Pull request by nfagerlund**
Per the resolution of https://github.com/hashicorp/terraform-website/pull/1573, we can now get rid of references to the old internally-hosted provider docs! Heck yeah. 

* This'll want a backport to 1.0. 
* Before this lands on stable-website, we'll want to do a redirect on that index page in terraform-website. ~PR incoming shortly.~ https://github.com/hashicorp/terraform-website/pull/1768

### CODE REVIEW

**Comment by nfagerlund** [comment]
The link check failure is a false alarm 😑 and unfortunately I don't really have time to go fix that recurring problem right now. Please go ahead and merge-while-failing.

File: website/docs/configuration-0-11/data-sources.html.md [hunk 1]
```diff
@@ -64,8 +64,8 @@ parameter) and `NAME` (second parameter). The combination of the type
 and name must be unique.
 
 Within the block (the `{ }`) is configuration for the data instance. The
-configuration is dependent on the type, and is documented for each
-data source in the [providers section](/docs/providers/index.html).
+configuration is dependent on the type; consult the provider's documentation for
```
Comment (laurapacilio) [inline website/docs/configuration-0-11/data-sources.html.md:67]: Maybe we link to the registry here?

```suggestion
configuration is dependent on the type; consult the [provider's documentation] (https://registry.terraform.io/browse/providers) for
```

File: website/docs/configuration-0-11/providers.html.md [hunk 2]
```diff
@@ -48,8 +48,8 @@ header. For example, `provider "aws"` above is a configuration for the
 `aws` provider.
 
 Within the block body (between `{ }`) is configuration for the provider.
-The configuration is dependent on the type, and is documented
-[for each provider](/docs/providers/index.html).
+The configuration is dependent on the type, and is described
```
Comment (laurapacilio) [inline website/docs/configuration-0-11/providers.html.md:51]:
```suggestion
The configuration is dependent on the type. Consult the [provider's documentation](https://registry.terraform.io/browse/providers) for details. 
```

File: website/docs/configuration-0-11/resources.html.md [hunk 3]
```diff
@@ -41,9 +41,8 @@ parameter) and `NAME` (second parameter). The combination of the type
 and name must be unique.
 
 Within the block (the `{ }`) is configuration for the resource. The
-configuration is dependent on the type, and is documented for each
-resource type in the
-[providers section](/docs/providers/index.html).
+configuration is dependent on the type; consult the provider's documentation for
```
Comment (laurapacilio) [inline website/docs/configuration-0-11/resources.html.md:44]:
```suggestion
configuration is dependent on the type. Consult the [provider's documentation](https://registry.terraform.io/browse/providers) for details. 

```

File: website/docs/language/providers/index.html.md [hunk 4]
```diff
@@ -40,6 +40,21 @@ The [Terraform Registry](https://registry.terraform.io/browse/providers)
 is the main directory of publicly available Terraform providers, and hosts
 providers for most major infrastructure platforms.
 
+## Provider Documentation
+
+Each provider has its own documentation, describing its resource
+types and their arguments.
+
+The [Terraform Registry](https://registry.terraform.io/browse/providers)
+includes documentation for all of the providers it hosts. Use the
```
Comment (laurapacilio) [inline website/docs/language/providers/index.html.md:49]:
```suggestion
includes documentation for a wide range of providers developed by HashiCorp, third-party vendors, and our Terraform community. Use the
```

**Review by laurapacilio — APPROVED** [review]
Looks good - thank you for doing this! Left some suggestions :-)

**Pull request merged**